integration_tests/src/tests/offline_processing.rs

```diff
@@ -0,0 +1,171 @@
+
+use anyhow::{anyhow, Result};
+use crossbeam_channel::Sender;
+use futures_util::{SinkExt, StreamExt};
+use live_compositor::config::read_config;
+use log::info;
+use serde_json::json;
+use tokio_tungstenite::tungstenite;
+
+use crate::CompositorInstance;
+
+const BUNNY_URL: &str =
+    "https://commondatastorage.googleapis.com/gtv-videos-bucket/sample/BigBuckBunny.mp4";
+
+#[test]
+pub fn offline_processing() -> Result<()> {
+    const OUTPUT_FILE: &str = "/tmp/offline_processing_output.mp4";
+    let mut config = read_config();
+    config.queue_options.ahead_of_time_processing = true;
+    config.queue_options.never_drop_output_frames = true;
+    let instance = CompositorInstance::start(Some(config));
+    let (msg_sender, msg_receiver) = crossbeam_channel::unbounded();
+    start_server_msg_listener(instance.api_port, msg_sender);
+
+    instance.send_request(
+        "input/input_1/register",
+        json!({
+            "type": "mp4",
+            "url":  BUNNY_URL,
+            "required": true
+        }),
+    )?;
+
+    instance.send_request(
+        "output/output_1/register",
+        json!({
+            "type": "mp4",
+            "path": OUTPUT_FILE,
+            "video": {
+                "resolution": {
+                    "width": 640,
+                    "height": 320
+                },
+                "encoder": {
+                    "type": "ffmpeg_h264",
+                    "preset": "ultrafast",
+                },
+                "initial": {
+                    "root": {
+                       "type": "view",
+                       "children": [{
+                            "type": "rescaler",
+                            "child": {
+                                "type": "input_stream",
+                                "input_id": "input_1"
+                            }
+                        }]
+                    }
+                },
+                "send_eos_when": { "all_inputs": true }
+            },
+            "audio": {
+                "encoder": {
+                    "type": "aac",
+                    "channels": "stereo"
+                },
+                "initial": {
+                    "inputs": [{ "input_id": "input_1" }]
+                },
+                "send_eos_when": { "all_inputs": true }
+            }
+        }),
+    )?;
+
+    instance.send_request(
+        "input/input_1/unregister",
+        json!({
+            "schedule_time_ms": 2000
+        }),
+    )?;
+    instance.send_request(
+        "output/output_1/unregister",
+        json!({
+            "schedule_time_ms": 2000
+        }),
+    )?;
+
+    instance.send_request("start", json!({}))?;
+
+    for msg in msg_receiver.iter() {
+        if let tungstenite::Message::Text(msg) = msg {
+            if msg.contains("\"type\":\"OUTPUT_DONE\",\"output_id\":\"output_1\"") {
+                info!("breaking");
+                break;
+            }
+        }
+    }
+
+    Command::new("ffprobe")
```

Comment:
Can we get duration or bitrate with regexp from the output 

Sth like this should be easy to match, but I'm not sure if ffmpeg output is consistent enough
```
  Duration: 00:09:56.47, start: 0.000000, bitrate: 2119 kb/s
```